presto-main/src/test/java/com/facebook/presto/server/TestServer.java

```diff
@@ -122,6 +124,37 @@ public void testServerStarts()
         assertEquals(response.getStatusCode(), OK.getStatusCode());
     }
 
+    @Test(timeOut = 10000)
+    public void testServerStartsWithGC()
+            throws Exception
+    {
+        // force a GC
+        long gcCount = getNumGcs();
```

Comment:
Once committer review time comes I think you will have to make this more black box (look up memory limits and then allocate to trigger full GC) or remove the test.